src/Hooks/GutenbergCleanup.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace WPML\PB\Elementor\Hooks;
+
+use WPML\FP\Fns;
+use WPML\FP\Maybe;
+use WPML\FP\Obj;
+use WPML\FP\Relation;
+use WPML\PB\Elementor\DataConvert;
+use WPML_Elementor_Data_Settings;
+use function WPML\FP\partial;
+use function WPML\FP\pipe;
+
+class GutenbergCleanup implements \IWPML_Backend_Action, \IWPML_Frontend_Action, \IWPML_DIC_Action {
+
+	public function add_hooks() {
+		add_filter(
+			'update_post_metadata',
+			Fns::withoutRecursion( Fns::identity(), [ $this, 'removeGutenbergFootprint' ] ),
+			10,
+			4
+		);
+	}
+
+	/**
+	 * If we detect Gutenberg footprint in the Elementor data,
+	 * we'll remove it and delete the Gutenberg string package.
+	 *
+	 * @param null|bool $check
+	 * @param int       $postId
+	 * @param string    $metaKey
+	 * @param string    $metaValue
+	 *
+	 * @return mixed
+	 */
+	public function removeGutenbergFootprint( $check, $postId, $metaKey, $metaValue ) {
+		if (
+			WPML_Elementor_Data_Settings::META_KEY_DATA === $metaKey
+			&& WPML_Elementor_Data_Settings::is_edited_with_elementor( $postId )
+		) {
+			$data = DataConvert::unserialize( $metaValue );
+
+			// $ifDataHasChanged :: array -> bool
+			$ifDataHasChanged = function( $processedData ) use ( $data ) {
+				return json_encode( $processedData ) !== json_encode( $data );
+			};
+
+			// $updatePostMeta :: array -> array
+			$updatePostMeta = Fns::tap( pipe(
+				[ DataConvert::class, 'serialize' ],
+				partial( 'update_metadata', 'post', $postId, WPML_Elementor_Data_Settings::META_KEY_DATA )
+			) );
+
+			// $deleteGbPackage :: array -> array
+			$deleteGbPackage = Fns::tap( pipe(
+				Fns::always( $postId ),
+				[ self::class, 'getGutenbergPackage' ],
+				[ self::class, 'deletePackage' ]
+			) );
+
+			return Maybe::of( $data )
+				->map( [ $this, 'removeBlockMetaInEditorWidget' ] )
+				->filter( $ifDataHasChanged )
+				->map( $updatePostMeta )
+				->map( $deleteGbPackage )
+				->map( Fns::always( true ) )
+				->getOrElse( $check );
+		}
+
+		return $check;
+	}
+
+	/**
+	 * @param array $data
+	 *
+	 * @return array
+	 */
+	public function removeBlockMetaInEditorWidget( array $data ) {
+		foreach ( $data as &$element ) {
+			if ( $element['elements'] ) {
+				$element['elements'] = $this->removeBlockMetaInEditorWidget( $element['elements'] );
+			} elseif ( 'widget' === $element['elType'] && isset( $element['settings']['editor'] ) ) {
+				$element['settings']['editor'] = preg_replace( '(<!--[^<]*-->)', '', $element['settings']['editor'] );
+			}
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @param int $postId
+	 */
+	public static function getGutenbergPackage( $postId ) {
+		// $isGbPackage :: \WPML_Package -> bool
+		$isGbPackage = pipe( Obj::prop( 'kind_slug' ), Relation::equals( 'gutenberg' ) );
+
+		return wpml_collect( apply_filters( 'wpml_st_get_post_string_packages', [], $postId ) )
+			->filter( $isGbPackage )
```

Comment:
@strategio You can use Relation::propEq here.
```php
->filter( Relation::propEq( 'kind_slug', 'gutenberg' ) )
```

Reply:
Thanks @brucepearson (I will keep the closure variable, it make the logic more meaningfull to me).